pkg/controller/infrastructure/terraform.go

```diff
@@ -64,12 +64,21 @@ func (terraformOps) ComputeChartValues(
 		if workersCIDR == "" {
 			workersCIDR = zone.Worker
 		}
-		zones = append(zones, map[string]interface{}{
+
+		zoneConfig := map[string]interface{}{
 			"name": zone.Name,
 			"cidr": map[string]interface{}{
 				"workers": string(workersCIDR),
 			},
-		})
+		}
+
+		if zone.NatGateway != nil && zone.NatGateway.IPAddress != nil {
+			natGateway := make(map[string]interface{})
+			natGateway["ipAddress"] = string(*zone.NatGateway.IPAddress)
+			zoneConfig["natGateway"] = natGateway
```

Comment:
The `string()` should not be needed. Also, I'd write it this way for better readability (though, that's just my personal preference ;-)):
```suggestion
			zoneConfig["natGateway"] = map[string]interface{}{
					"ipAddress": *zone.NatGateway.IPAddress,
			}
```